mllib/src/main/scala/org/apache/spark/ml/feature/QuantileDiscretizer.scala

```diff
@@ -114,10 +115,10 @@ final class QuantileDiscretizer @Since("1.6.0") (@Since("1.6.0") override val ui
     splits(0) = Double.NegativeInfinity
     splits(splits.length - 1) = Double.PositiveInfinity
 
-    val distinctSplits = splits.distinct
+    val distinctSplits = splits.filter(!_.isNaN).distinct
```

Comment:
What would you do with NaN in approxQuantile except ignore it? it can't affect a quantile. The question of what the bucketizer should do with NaN is separate. I'd still favor one behavior there, but keeping that separate, let's figure out what approxQuantile should do.